commentwriter.go

```diff
@@ -0,0 +1,42 @@
+package vfsgen
+
+import "io"
+
+// commentWriter writes given input to underlying io.Writer as a Go comment,
+// using line comments (//).
+type commentWriter struct {
+	io.Writer
+
+	wroteComment bool
+}
+
+func (cw *commentWriter) Write(p []byte) (n int, err error) {
+	for i, c := range p {
+		if c == '\n' {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("//"))
+				if err != nil {
+					return n, err
+				}
+			}
+			cw.wroteComment = false
+		} else {
+			if !cw.wroteComment {
+				_, err := cw.Writer.Write([]byte("// "))
+				if err != nil {
+					return n, err
+				}
+				cw.wroteComment = true
+			}
+		}
+		_, err := cw.Writer.Write(p[i : i+1])
+		if err != nil {
+			return n, err
+		}
+		n++
+	}
+	if n != len(p) {
+		return n, io.ErrShortWrite
+	}
```

Comment:
> Since the final check on n is unnecessary, you might consider also incrementing n by 2 when you write an empty comment and 3 when you write the beginning of a comment line to reflect the actual bytes you've written, in case this is useful to the caller in debugging.

Can't do that, as it would break the [`io.Writer`](https://godoc.org/io#Writer) interface contract:

> Write writes len(p) bytes from p to the underlying data stream. It returns the number of bytes written from p (0 <= n <= len(p))

The other suggestions are good, I'll take them, thanks.